what4/src/What4/Expr/Builder.hs

```diff
@@ -1639,6 +1640,7 @@ instance IsExprBuilder (ExprBuilder t st fs) where
 
   getAnnotation _sym e =
     case e of
+      BoundVarExpr (bvarId -> n) -> Just n
       NonceAppExpr (nonceExprApp -> Annotation _ n _) -> Just n
       _ -> Nothing
 
```

Comment:
Note that I did *not* change `getUnannotatedTerm` to return `Just` for `BoundVarExpr`s. Let me know if anyone thinks I should.

Reply:
I'm inclined to make the implementations of `annotateTerm` and `getUnannotatedTerm` also special-case `BoundVarExpr`, as doing so feels like it would avoid some potential surprising behavior down the line.

Reply:
Done! Let me know if the implementation looks unsurprising.